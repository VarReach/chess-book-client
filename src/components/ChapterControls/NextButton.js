import React, { Component } from 'react';
import UserContext from '../../contexts/UserContext';
import { Link } from 'react-router-dom';

export default class NextButton extends Component {
  static contextType = UserContext;

  render() {
    return (
      <Link
          className="article__chapter-next-button"
          to={"/"}
          onClick={(e) => {
              this.props.handleOnClick(e, this.props.bookId, this.props.chapterIndex + 1);
              this.context.completeChapter(this.props.chapterId);
            }
          }
      >
        <i className='fas fa-angle-right'/>
      </Link>
    )
  }
}