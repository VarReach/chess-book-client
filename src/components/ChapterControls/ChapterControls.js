import React, { Component } from 'react';
import UserContext from '../../contexts/UserContext';
import { Link } from 'react-router-dom';
import NextButton from './NextButton';

export default class ChapterControls extends Component {
  static contextType = UserContext;

  state = {
    showScrollButton: null,
  }

  componentDidMount = () => {
    window.addEventListener('scroll', this.handleScroll);
  }

  componentWillUnmount = () => {
    window.removeEventListener('scroll', this.handleScroll);
  }

  handleScroll = () => {
    if (document.body.scrollTop > 20 || document.documentElement.scrollTop > 20){
      this.setState({ showScrollButton: true });
    } else {
      this.setState({ showScrollButton: null });
    }
  }

  scrollToTop = () => {
    document.body.scrollTop = 0;
    document.documentElement.scrollTop = 0;
  }

  render() {
    return (
      <div className="article__chapter-controls">
        <div className="article__chapter-buttons-holder">
          {this.props.chapterIndex === 1
            ? <button disabled={true}><i className='fas fa-angle-double-left'/></button>
            : <Link 
                to={`/book/${this.props.bookId}/chapter/1`} 
                onClick={(e) => this.props.handleControlsOnClick(e, this.props.bookId, 1)}>
                  <i className='fas fa-angle-double-left'/>
              </Link>}
          {this.props.chapterIndex === 1
            ? <button disabled={true}><i className='fas fa-angle-left'/></button>
            : <Link 
                to={`/book/${this.props.bookId}/chapter/1`} 
                onClick={(e) => this.props.handleControlsOnClick(e, this.props.bookId, this.props.chapterIndex - 1)}>
                  <i className='fas fa-angle-left'/>
              </Link>}
          <NextButton bookId={this.props.bookId} chapterId={this.props.chapterId} chapterIndex={this.props.chapterIndex} handleOnClick={this.props.handleControlsOnClick} />
        </div>
        {this.state.showScrollButton && <button onClick={this.scrollToTop}><i className="far fa-arrow-alt-circle-up"/></button>}
      </div>
    )
  }
}