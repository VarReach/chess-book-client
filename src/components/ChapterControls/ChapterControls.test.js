import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ChapterControls from './ChapterControls';

describe('ChapterControls component', () => {
  const props = {
    bookId: 1,
    chapterId: 1,
    chapterIndex: 1,
  }

  it('renders the controls', () => {
    const wrapper = shallow(<ChapterControls/>)
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders the controls given props', () => {
    const wrapper = shallow(<ChapterControls {...props}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});