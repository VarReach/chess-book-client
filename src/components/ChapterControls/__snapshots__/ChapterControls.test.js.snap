// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChapterControls component renders the controls 1`] = `
<div
  className="article__chapter-controls"
>
  <div
    className="article__chapter-buttons-holder"
  >
    <Link
      onClick={[Function]}
      replace={false}
      to="/book/undefined/chapter/1"
    >
      <i
        className="fas fa-angle-double-left"
      />
    </Link>
    <Link
      onClick={[Function]}
      replace={false}
      to="/book/undefined/chapter/1"
    >
      <i
        className="fas fa-angle-left"
      />
    </Link>
    <NextButton />
  </div>
</div>
`;

exports[`ChapterControls component renders the controls given props 1`] = `
<div
  className="article__chapter-controls"
>
  <div
    className="article__chapter-buttons-holder"
  >
    <button
      disabled={true}
    >
      <i
        className="fas fa-angle-double-left"
      />
    </button>
    <button
      disabled={true}
    >
      <i
        className="fas fa-angle-left"
      />
    </button>
    <NextButton
      bookId={1}
      chapterId={1}
      chapterIndex={1}
    />
  </div>
</div>
`;
